src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -1891,29 +1934,23 @@ void Creature::ForcedDespawn(uint32 timeMSToDespawn, Seconds const& forceRespawn
         return;
     }
 
-    if (forceRespawnTimer > Seconds::zero())
+    // do it before killing creature
+    DestroyForNearbyPlayers();
+
+    bool overrideRespawnTime = true;
+    if (IsAlive())
     {
-        if (IsAlive())
-        {
-            uint32 respawnDelay = m_respawnDelay;
-            uint32 corpseDelay = m_corpseDelay;
-            m_respawnDelay = forceRespawnTimer.count();
-            m_corpseDelay = 0;
-            setDeathState(JUST_DIED);
-            m_respawnDelay = respawnDelay;
-            m_corpseDelay = corpseDelay;
-        }
-        else
+        setDeathState(JUST_DIED);
+
+        if (forceRespawnTimer > Seconds::zero())
         {
```

Comment:
ForcedDespawn has been far different in master than in 335...
We should check if this i now ok